source/shared/core_sqlsrv.h

```diff
@@ -2040,7 +2040,21 @@ namespace core {
         SQLRETURN r;
         SQLSMALLINT num_cols;
         r = ::SQLNumResultCols( stmt->handle(), &num_cols );
-        
+
+        // Workaround for a bug in unixODBC: after SQLExecDirect returns SQL_NO_DATA,
+        // r = ::SQLNumResultCols( stmt->handle(), &num_cols );  
+        // returns r=-1 (SQL_ERROR) and error HY010 (Function sequence error)
+        // but it should have succeeded with r=0 (SQL_SUCCESS) and no error
+        // instead of throwing an exception, return 0 if the r=-1, stament has been executed, and has a HY010 error
+        // (HY010 error should not return if stmt->execute is true)
+#ifndef _WIN32
```

Comment:
What happens if you execute a large  batch of SQL statements, for example 10000 inserts? According to [this](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlnumresultcols-function) page, SQLNumResultCols() might return HY010 when executing function is still executing.